python/cucim/src/cucim/core/operations/color/stain_normalizer.py

```diff
@@ -0,0 +1,821 @@
+        be within unsigned 8-bit integer intensity range ([0, 255]) when
+        ``image_type == 'intensity'``.
+    source_intensity : float, optional
+        Transmitted light intensity. The algorithm will clip image intensities
+        above the specified `source_intensity` and then normalize by
+        `source_intensity` so that `image` intensities are <= 1.0. Only used
+        when `image_type=='intensity'`.
+    alpha : float, optional
+        Algorithm parameter controlling the ``[alpha, 100 - alpha]``
+        percentile range used as a robust [min, max] estimate.
+    beta : float, optional
+        Absorbance (optical density) threshold below which to consider pixels
+        as transparent. Transparent pixels are excluded from the estimation.
+    ref_stain_coeff : array-like
+        Reference stain coefficients as determined by the output of
+        `stain_decomposition` for a reference image.
+    ref_max_conc : tuple or cp.ndarray
+        The reference maximum concentrations.
+    image_type : {'intensity', 'absorbance'}, optional
+        With the default `image_type` of `'intensity'`, the image will be
+        transformed to an `absorbance` using ``image_to_absorbance``. If
+        the input `image` is already an absorbance image, then `image_type`
+        should be set to `'absorbance'` instead.
+    channel_axis : int, optional
+        The axis corresponding to color channels (default is the last axis).
+    method : {'ortho', 'lstsq'}, optional
+        The method used to solve for the raw stain concentrations. 'ortho'
+        the inverse of the source stain coefficients matrix by the absorbances.
+        This can be used when the source stain coefficients form a square
+        matrix. `lstsq` performs a least-squares solution, and can be used when
+        there are fewer stain vectors than color channels.
+
+    Returns
+    -------
+    stain_coeff : np.ndarray
+        Stain attenuation coefficient matrix derived from the image, where
+        the first column corresponds to H, the second column is E and the rows
+        are RGB values.
+
+    References
+    ----------
+    .. [1] M. Macenko et al., "A method for normalizing histology slides for
+           quantitative analysis," 2009 IEEE International Symposium on
+           Biomedical Imaging: From Nano to Macro, 2009, pp. 1107-1110,
+           doi: 10.1109/ISBI.2009.5193250.
+           http://wwwx.cs.unc.edu/~mn/sites/default/files/macenko2009.pdf
+    """
+
+    _validate_image(image)
+    channel_axis = _prep_channel_axis(channel_axis, image.ndim)
+
+    # convert to absorbance (optical density) matrix of shape (n_pixels, 3)
+    absorbance = _image_to_absorbance_matrix(
+        image,
+        source_intensity=source_intensity,
+        image_type=image_type,
+        channel_axis=channel_axis
+    )
+
+    # channels_axis=0 for the shape (3, n_pixels) absorbance matrix
+    src_stain_coeff = stain_decomposition_macenko(
```

Comment:
should perhaps change "decomposition"->"extraction". not sure why I had named it this way

Reply:
Both are fine to me since you need to decompose stains to extract them.
However, speaking on naming, do you think that we can give it a more representative name to this method? Rather than the family name of the first author of that paper, maybe we can come use the name of underlying method, since it is not a new method per se, and the paper just showed how they have used this method in histopathology.